main/src/main/scala/sbt/Keys.scala

```diff
@@ -456,6 +456,17 @@ object Keys {
   val (executionRoots, dummyRoots) = Def.dummy[Seq[ScopedKey[_]]]("executionRoots", "The list of root tasks for this task execution.  Roots are the top-level tasks that were directly requested to be run.")
   val state = Def.stateKey
   val streamsManager = Def.streamsManagerKey
+  // wrapper to work around SI-2915
+  final class TaskProgress(val progress: ExecuteProgress[Task])
+  object TaskProgress {
+    def apply(progress: ExecuteProgress[Task]): TaskProgress = new TaskProgress(progress)
+  }
+  val taskProgress = settingKey[Boolean]("Enables (true) or disables the super shell.")
+  val progressReports = settingKey[State => Seq[TaskProgress]]("Task progress reporters.").withRank(DTask)
```

Comment:
It isn't really clear to me at a glance exactly what `progressReports` does or how it's connected to the `taskProgress` key. I wouldn't mind a comment or a more detailed key description.

Reply:
ok.